### PR TITLE
feat(rbac): fetch org role during permission resolution

### LINKED ISSUE #2011: FEAT: Fetch org role during permission resolution
## Problem
`hasProjectPermission()` (and `hasTeamPermission()`) fetches the `TeamUser` record to resolve the team role, but never fetches `OrganizationUser.role`. The EXTERNAL org role (lite member) is invisible during project-level permission checks, so we can't distinguish lite members from regular VIEWERs on the server side.

## Proposed Solution
Extend the Prisma query in `hasProjectPermission()` and `hasTeamPermission()` to include `OrganizationUser.role` via the team → organization relationship. Attach the org role to the permission context so downstream code can read it. Expose that org role to the frontend via `useOrganizationTeamProject()`.

**Details:**
1. Include `organizationRole` in the Prisma include when loading team membership.
2. Return the org role in the permission resolution result without altering current behavior.
3. Mirror the change in `hasTeamPermission()` if it also lacks the org role.
4. Update `useOrganizationTeamProject()` to expose `organizationRole` and make it available to route guards.
5. Add tests proving the org role is fetched and that existing permissions (ADMIN, MEMBER, VIEWER) remain unchanged.

## Acceptance Criteria
- [ ] `hasProjectPermission()` returns the user's org role alongside the team role
- [ ] `hasTeamPermission()` also returns org role if it previously did not
- [ ] No permission check behavior changes — same inputs produce the same allow/deny results
- [ ] Frontend `useOrganizationTeamProject()` hook exposes the user's org role
- [ ] Unit tests pass for all existing team roles (ADMIN, MEMBER, VIEWER, CUSTOM)
- [ ] Unit tests verify org role is available for EXTERNAL, MEMBER, ADMIN org roles

## Context
Planning doc: `EPIC/how-to/lite-members/epic_issues.md`
Parent epic: langwatch/langwatch-saas#315
Spec: `specs/licensing/lite-member-restrictions.feature` (TBD)

### PR DESCRIPTION
**Pull request by sergioestebance**
## Summary

- New `resolveProjectPermission()` and `resolveTeamPermission()` functions that return `PermissionResult { permitted, organizationRole }`
- Existing `hasProjectPermission`/`hasTeamPermission` delegate to them — still return `boolean`, zero blast radius to 60+ callers
- Middleware `checkProjectPermission`/`checkTeamPermission` attach `organizationRole` to tRPC context
- Frontend hook `useOrganizationTeamProject()` exposes `organizationRole`
- 63 integration tests covering all org role types, demo projects, custom roles, regression matrix

Closes #2011

> **Note:** The EXTERNAL → LITE_MEMBER rename (#2019) was originally included but has been deferred to a follow-up PR due to on-prem deployment safety concerns (code/migration ordering).

## Test plan

- [x] All 63 integration tests pass
- [x] Typecheck clean
- [x] Boolean API regression guards (hasProjectPermission/hasTeamPermission still return boolean)
- [x] Middleware propagates organizationRole to ctx
- [x] Demo project edge case handled

### CODE REVIEW

File: langwatch/src/server/api/__tests__/rbac-integration.test.ts [hunk 1]
```diff
@@ -573,4 +584,637 @@ describe("RBAC Integration Tests", () => {
...
+          { prisma: mockPrisma, session: mockSession },
+          "demo-project-1",
+          "analytics:view" as Permission,
+        );
+
+        expect(result.permitted).toBe(true);
+        expect(result.organizationRole).toBeNull();
+
+        delete process.env.DEMO_PROJECT_ID;
+      });
```
[the remark this answers is not in the document]
Comment (sergioestebance) [reply langwatch/src/server/api/__tests__/rbac-integration.test.ts:700]: Already fixed in commit fb82d88 — the `DEMO_PROJECT_ID` env var is now wrapped in `try/finally` (lines 688-699).

File: langwatch/src/server/api/__tests__/rbac-integration.test.ts [hunk 2]
```diff
@@ -573,4 +584,638 @@ describe("RBAC Integration Tests", () => {
...
+          { prisma: mockPrisma, session: mockSession },
+          "team-1",
+          "team:view" as Permission,
+        );
+
+        expect(result.permitted).toBe(false);
+        expect(result.organizationRole).toBeNull();
+      });
+    });
+  });
```
[the remark this answers is not in the document]
Comment (sergioestebance) [reply langwatch/src/server/api/__tests__/rbac-integration.test.ts:935]: Added in commit 5cf6bc3 — two CUSTOM role test cases (grant + deny) for `resolveTeamPermission`, matching the existing coverage for `resolveProjectPermission`.
